src/Evolution/Systems/Cce/ReadBoundaryDataH5.cpp

```diff
@@ -226,7 +230,127 @@ void SpecWorldtubeH5BufferUpdater::update_buffer(
   }
 }
 
+ReducedSpecWorldtubeH5BufferUpdater::ReducedSpecWorldtubeH5BufferUpdater(
+    const std::string& cce_data_filename) noexcept
+    : cce_data_file_{cce_data_filename}, filename_{cce_data_filename} {
+  get<Tags::detail::InputDataSet<
+      Spectral::Swsh::Tags::SwshTransform<Tags::BondiBeta>>>(dataset_names_) =
+      "Beta";
+  get<Tags::detail::InputDataSet<
+      Spectral::Swsh::Tags::SwshTransform<Tags::BondiU>>>(dataset_names_) = "U";
+  get<Tags::detail::InputDataSet<
+      Spectral::Swsh::Tags::SwshTransform<Tags::BondiQ>>>(dataset_names_) = "Q";
+  get<Tags::detail::InputDataSet<
+      Spectral::Swsh::Tags::SwshTransform<Tags::BondiW>>>(dataset_names_) = "W";
+  get<Tags::detail::InputDataSet<
+      Spectral::Swsh::Tags::SwshTransform<Tags::BondiJ>>>(dataset_names_) = "J";
+  get<Tags::detail::InputDataSet<
+      Spectral::Swsh::Tags::SwshTransform<Tags::Dr<Tags::BondiJ>>>>(
+      dataset_names_) = "DrJ";
+  get<Tags::detail::InputDataSet<
+      Spectral::Swsh::Tags::SwshTransform<Tags::BondiH>>>(dataset_names_) = "H";
+  get<Tags::detail::InputDataSet<
+      Spectral::Swsh::Tags::SwshTransform<Tags::BondiR>>>(dataset_names_) = "R";
+  get<Tags::detail::InputDataSet<
+      Spectral::Swsh::Tags::SwshTransform<Tags::DuRDividedByR>>>(
+      dataset_names_) = "DuRDividedByR";
+
+  // We assume that the filename has the extraction radius encoded as an
+  // integer between the first occurrence of 'R' and the first occurrence of
+  // '.'. This is the format provided by SpEC.
+  const size_t r_pos = cce_data_filename.find('R');
+  const size_t dot_pos = cce_data_filename.find('.');
+  const std::string text_radius =
+      cce_data_filename.substr(r_pos + 1, dot_pos - r_pos - 1);
+  extraction_radius_ = stod(text_radius);
```

Comment:
the `stod` should throw an `invalid_argument` exception, which should cause the program to terminate because the function is `noexcept` would you like a catch here with a more detailed `ERROR`?

Reply:
Yes, that would be a lot easier for users.